core/clustersmngr/clustersmngr.go

```diff
@@ -72,14 +74,19 @@ type clientsPool struct {
 	mutex   sync.Mutex
 }
 
-type ClusterClientConfig func(Cluster) *rest.Config
+type ClusterClientConfig func(Cluster) (*rest.Config, error)
 
 func ClientConfigWithUser(user *auth.UserPrincipal) ClusterClientConfig {
-	return func(cluster Cluster) *rest.Config {
+	return func(cluster Cluster) (*rest.Config, error) {
 		config := &rest.Config{
 			Host:            cluster.Server,
 			TLSClientConfig: cluster.TLSConfig,
 			Timeout:         kubeClientTimeout,
+			Dial: (&net.Dialer{
+				Timeout: kubeClientDialTimeout,
```

Comment:
Presumably this ends up in https://pkg.go.dev/net#DialTimeout

Under the hood I wonder if this is effecting every request? `roundtripper.IsEnabled` doesn't use dial explicitly, its just doing a HEAD etc.

Maybe 3s is a bit low unfortunately?